edx_sga/sga.py

```diff
@@ -0,0 +1,530 @@
+        default=None,
+        help="When the annotated file was uploaded")
+
+    def max_score(self):
+        return self.points
+
+    @reify
+    def block_id(self):
+        # cargo culted gibberish
+        return self.scope_ids.usage_id
+
+    def student_submission_id(self, id=None):
+        """
+        Returns dict required by the submissions app for creating and
+        retrieving submissions for a particular student.
+        """
+        if id is None:
+            id = self.xmodule_runtime.anonymous_student_id
+            assert id != 'MOCK', "Forgot to call 'personalize' in test."
+        return {
+            "student_id": id,
+            "course_id": self.course_id,
+            "item_id": self.block_id,
+            "item_type": 'sga',  # ???
+        }
+
+    def get_submission(self, id=None):
+        """
+        Get student's most recent submission.
+        """
+        submissions = submissions_api.get_submissions(
+            self.student_submission_id(id))
+        if submissions:
+            # If I understand docs correctly, most recent submission should
+            # be first
+            return submissions[0]
+
+    def get_score(self, id=None):
+        """
+        Get student's current score.
+        """
+        score = submissions_api.get_score(self.student_submission_id(id))
+        if score:
+            return score['points_earned']
+
+    @reify
+    def score(self):
+        return self.get_score()
+
+    def student_view(self, context=None):
+        """
+        The primary view of the StaffGradedAssignmentXBlock, shown to students
+        when viewing courses.
+        """
+        context = {
+            "student_state": json.dumps(self.student_state()),
+            "id": self.location.name.replace('.', '_')
+        }
+        if self.show_staff_grading_interface():
+            context['is_course_staff'] = True
+            self.update_staff_debug_context(context)
+
+        fragment = Fragment()
+        fragment.add_content(
+            render_template(
+                'templates/staff_graded_assignment/show.html',
+                context
+            )
+        )
+        fragment.add_css(_resource("static/css/edx_sga.css"))
+        fragment.add_javascript(_resource("static/js/src/edx_sga.js"))
+        fragment.initialize_js('StaffGradedAssignmentXBlock')
+        return fragment
+
+    def update_staff_debug_context(self, context):
+        published = self.start
+        context['is_released'] = published and published < _now()
+        context['location'] = self.location
+        context['category'] = type(self).__name__
+        context['fields'] = [
+            (name, field.read_from(self))
+            for name, field in self.fields.items()]
+
+    def student_state(self):
+        """
+        Returns a JSON serializable representation of student's state for
+        rendering in client view.
+        """
+        submission = self.get_submission()
+        if submission:
```

Comment:
Personal nitpick, you can decide to ignore it: You may reduce branch complexity on this function if you declare your variables at the beginning

uploaded = None
annotated = None
graded = None

Then you'd just drop all your else statements. Just personal preference, though.